packages/firestore/src/api/database.ts

```diff
@@ -428,18 +425,24 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
   }
 
   _clearPersistence(): Promise<void> {
-    if (this.clientRunning) {
-      throw new FirestoreError(
-        Code.FAILED_PRECONDITION,
-        'Persistence cannot be cleared while the client is running'
-      );
-    }
     const persistenceKey = IndexedDbPersistence.buildStoragePrefix(
       this.makeDatabaseInfo()
     );
     const deferred = new Deferred<void>();
     this._queue.enqueueAndForget(async () => {
       try {
+        if (
+          this._firestoreClient !== undefined &&
+          !this._firestoreClient.isShutdown()
+        ) {
+          deferred.reject(
```

Comment:
Since you're already in a `try`/`catch` block you could accomplish the same without an explicit reject/return. You could just `throw new FirestoreError(...)` and allow the `catch` to perform the rejection for you.